### ISSUE TITLE
Bug: Collada (dae) import fails on SceneKit exported scene

### ISSUE THREAD

**Issue by chinery** [assimp/assimp#4038]
**Describe the bug**
I am using `assimp_cmd` to try to convert .dae format scenes that have been exported from SceneKit in mac OS. The .dae files open perfectly fine in Blender. But the converted files are broken: the faces are disconnected, and something is wrong with front/back faces (winding order?). Also, when I convert to .fbx format, the scale does not seem to be set properly (model comes in very small), but I've manually corrected that in the images below.

I've attached this sample dae file and the converted fbx I get as a result
[scenekit dae.zip](https://github.com/assimp/assimp/files/7015641/scenekit.dae.zip)

**To Reproduce**
1. Extract files in .zip above
2. Run `assimp export scenekit.dae scenekit.fbx`
3. Inspect files in graphics editor

**Expected behavior**
The two models should look the same

**Screenshots**
DAE file on the left, converted FBX file on the right.
<img width="845" alt="Screenshot 2021-08-19 at 16 27 01" src="https://user-images.githubusercontent.com/8233798/130096884-2faa9495-0d68-4a9f-beb5-f1bdbef26215.png">
<img width="830" alt="Screenshot 2021-08-19 at 16 27 11" src="https://user-images.githubusercontent.com/8233798/130096897-07aef8de-f453-4841-976b-8fe3ebb7260b.png">

**Platform (please complete the following information):**
 - OS: mac OS 11.2.3
 - assimp version: Version 5.0 -shared -st (GIT commit ceced70d)

**Additional context**
I've tested this on an Intel and M1 mac and get the same results. Also, I tried converting the duck.dae model from the examples directory, and it works fine. So I assume SceneKit must be doing something differently in its Collada exporter, which Blender can account for, but assimp cannot. Thanks!

**Comment by chinery** [comment]
I think I've found part of the issue. SceneKit exports its normals inside the `<vertices>` element as well as in the face description, line 110 of the previously attached .dae file has:

```
    <vertices id="geometrySource4-vertices">
     <input semantic="POSITION" source="#geometrySource4"/>
     <input semantic="NORMAL" source="#geometrySource6"/>
    </vertices>
```

If you delete line 110 which has the NORMAL semantic, then run the assimp convert command, the result looks good without textures:
(dae left, fbx right)
<img width="952" alt="Screenshot 2021-08-20 at 12 21 36" src="https://user-images.githubusercontent.com/8233798/130226032-b54c4d13-ca00-4ede-bd7b-c8a11657be29.png">

But there still seems to be some depth issue when viewing with textures on
<img width="951" alt="Screenshot 2021-08-20 at 12 21 49" src="https://user-images.githubusercontent.com/8233798/130226080-563b8589-d641-4227-8808-9096bfd0d248.png">

**Comment by chinery** [comment]
Someone just helpfully pointed out to me that when you open this fbx file in Blender, it sets the "Blend Mode" setting to "Alpha Blend". Setting it to _any_ other setting fixes the partially transparent textures (Opaque, Alpha Clip (with alpha < 1.0), or Alpha Hashed).

<img width="311" alt="Screenshot 2021-08-20 at 19 38 58" src="https://user-images.githubusercontent.com/8233798/130279752-62246233-1d74-48c4-8da8-63e5ecbc1fff.png">

When I load the duck.fbx file I converted from the sample dae files, I notice that it sets Blend Mode to Opaque (and if I set it to Alpha Blend, it looks bad again). Is there something (or something missing) in the Collada file, or a flag for assimp maybe? Any advice appreciated.

**Comment by chinery** [comment]
Update: I think the texture rendering issue is Blender's. When you import an fbx file that has a png texture it sets the blend mode to alpha blend, no matter whether the texture contains any transparency. The reason I was confused before is because the duck.dae sample file has a .tga texture.

So in summary:
- SceneKit exports its vertex normals into the `<vertices>` element _and_ the primitive element (`<triangles>`), which causes problems with Assimp. I'm not sure if the Collada spec rules on this either way, but Blender handles it fine as input.
- Assimp does not seem to apply the scale correctly from the Collada file. The SceneKit ship loads as 9.79m wide from the original dae file, but the fbx loads as 0.0979m. The sample duck.dae file loads correctly, but it specifies `<unit meter="0.01" name="centimeter"/>` whereas the SceneKit ship has `<unit meter="1.000000"/>`. If I change the unit inside the duck file and convert it, the fbx file still loads in the same size.

**Comment by RichardTea** [comment]
Yes, Assimp will need to discard one of the duplicate inputs - that's what Blender is doing.

The trouble is, which one?

Easiest is to read the first, then discard the second (and subsequent etc) duplicate vertex semantic and log a warning about this.

Does that work for your exported SceneKit models?

ISO Collada doesn't explicitly forbid duplicate vertex data, so I guess that technically means it's permitted - Apple have a lot of prior form on intentionally implementing standards using the most obtuse and obstructive way of interpreting the text.

**Comment by chinery** [comment]
Yes that works fine for the SceneKit models. Since the duplicated normals point to the same source element it doesn't seem to matter which channel they are in in the Collada file, if you delete either of them then the Assimp conversion works.

Thanks!

**Comment by tellypresence** [comment]
Please add label "Export" to this issue

**Comment by chinery** [comment]
@tellypresence hi, do you know if there was ever a fix (or rather robustness for bad input) in assimp?

At least for me I no longer have need for this. And .dae export from SceneKit seemed to be a hardly documented/supported feature in 2021 (I'm not sure if that has changed).

**Comment by tellypresence** [comment]
@chinery Thank you for following up

> do you know if there was ever a fix (or rather robustness for bad input) in assimp?
That's an ongoing battle.  Problems are getting fixed but it's a slow process

> At least for me I no longer have need for this. And .dae export from SceneKit seemed to be a hardly documented/supported feature in 2021 (I'm not sure if that has changed).
Will close this for now
